src/main/java/pl/put/poznan/transformer/rest/JSONToolsController.java

```diff
@@ -5,96 +5,132 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-import pl.put.poznan.transformer.logic.JSONToolsService;
+import pl.put.poznan.transformer.logic.JSONShowDIff;
+import pl.put.poznan.transformer.logic.domain.JSONException;
+import pl.put.poznan.transformer.logic.domain.JSONObject;
+import pl.put.poznan.transformer.logic.tools.JSONToolBlacklist;
+import pl.put.poznan.transformer.logic.tools.JSONToolMinify;
+import pl.put.poznan.transformer.logic.tools.JSONToolPrettify;
+import pl.put.poznan.transformer.logic.tools.JSONToolWhitelist;
 
 import java.util.List;
 import java.util.Map;
 
+import static pl.put.poznan.transformer.logic.tools.JSONToolBox.*;
 
 @RestController
 @CrossOrigin
 @RequestMapping("/api/v1/")
 public class JSONToolsController {
     private static final Logger logger = LoggerFactory.getLogger(JSONToolsController.class);
-    private final ObjectMapper objectMapper = new ObjectMapper();
-    private final JSONToolsService service = new JSONToolsService();
 
     @RequestMapping(value="minify", method = { RequestMethod.POST }, produces = "application/json")
     public ResponseEntity<String> minify(
         @RequestBody String payload
     ) {
+        final JSONToolMinify tool = new JSONToolMinify();
+
         logger.info("POST /api/v1/minify");
-        JsonNode jsonNode = service.parse(payload);
-        if(jsonNode == null){
+        JsonNode jsonNode;
+
+        try{
+            jsonNode = parse(payload);
+        }catch(JSONException exception){
             return ResponseEntity.status(400).body("{\"error\":\"Payload is not valid JSON!\"}");
-        }else{
-            JsonNode json = service.getInput(jsonNode);
-            if(json == null){
-                return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");
-            }
-            if(jsonNode.size() != 1){
-                return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
-            }
-            try{
-                return ResponseEntity.status(200).body(service.minify(json));
-            }catch(Exception exception){
-                logger.error(exception.toString());
-                return ResponseEntity.status(500).body("{\"error\":\"Internal error!\"}");
-            }
+        }
+
+        if(jsonNode.size() != 1){
+            return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
```

Comment:
Pododawaj log przed returnami tak żeby wypisywało co się dzieje, tutaj masz co to jest slf4j:
https://www.baeldung.com/slf4j-with-log4j2-logback
a tutaj przykładowe użycie prze zemnie w naszym kodzie:
https://github.com/maciejbilinski/io-json-tools/blob/develop/src/main/java/pl/put/poznan/transformer/logic/tools/JSONToolMinify.java

Reply:
teraz zauważyłem że w innych miejscach zrobiłeś logi tylko tu zapomniałeś :)